src/lib/containers/QueryWrapper.tsx

```diff
@@ -37,6 +37,8 @@ export type QueryWrapperProps = {
   showBarChart?: boolean
   componentIndex?: number //used for deep linking
   shouldDeepLink?: boolean
+  onQueryChange?: (newQueryJson: string) => void
```

Comment:
If deep linking, provide way to inform caller that the Query has been updated (so you can update the URL).